src/lib/png/png.c

```diff
@@ -0,0 +1,976 @@
+		printf("Error : invalid IHDR chunk.\n");
+		return -1;
+	}
+
+	if (read_IHDR(fd, &png_head) != 0) {
+		return -1;
+	}
+
+	png_data->width = png_head.width;
+	png_data->height = png_head.height;
+	png_data->color = png_head.color_type;
+	png_data->image_size = png_head.width * png_head.height * png_head.color_type;
+	/* malloc mem for image */
+	png_data->image = (uint8_t*)malloc(png_data->image_size);
+
+	/**
+	 * Reading chunks until IEND
+	 * its simple version and support just important chunks
+	 * fun read_chunk is a stub
+	 * TODO add reading of other chunks
+	 */
+	while (true) {
+		lseek(fd, pos, SEEK_SET);
+		read(fd, &chunk_length, sizeof(chunk_length));
+		read(fd, chunk_name, sizeof(chunk_name));
+
+		if (memcmp(chunk_name, "IDAT", 4) == 0) {
+			/* read, decode IDAT and fill image buffer */
+			if (read_IDAT(fd, png_data->image, (int)htonl(chunk_length), &png_head) != 0) {
+				printf("Error while read and decode IDAT.\n");
+				free(png_data->image);
+				return -1;
+			}
+			flag_IDAT = 1;
+		} else if (memcmp(chunk_name, "IEND", 4) == 0) {
+			if (!flag_IDAT) {
+				printf("Error : there is no IDAT chunk in the file\n");
+				free(png_data->image);
+				return -1;
+			}
+			break;
+		} else if (memcmp(chunk_name, "bKGD", 4) == 0) {
+			read_chunk(fd, chunk_name);
+		} else if (memcmp(chunk_name, "cHRM", 4) == 0) {
+			read_chunk(fd, chunk_name);
+		} else if (memcmp(chunk_name, "dSIG", 4) == 0) {
+			read_chunk(fd, chunk_name);
+		} else if (memcmp(chunk_name, "eXIF", 4) == 0) {
+			read_chunk(fd, chunk_name);
+		} else if (memcmp(chunk_name, "gAMA", 4) == 0) {
+			read_chunk(fd, chunk_name);
+		} else if (memcmp(chunk_name, "hIST", 4) == 0) {
+			read_chunk(fd, chunk_name);
+		} else if (memcmp(chunk_name, "iCCP", 4) == 0) {
+			read_chunk(fd, chunk_name);
+		} else if (memcmp(chunk_name, "iTXt", 4) == 0) {
+			read_chunk(fd, chunk_name);
+		} else if (memcmp(chunk_name, "pHYs", 4) == 0) {
+			read_chunk(fd, chunk_name);
+		} else if (memcmp(chunk_name, "sBIT", 4) == 0) {
+			read_chunk(fd, chunk_name);
+		} else if (memcmp(chunk_name, "sPLT", 4) == 0) {
+			read_chunk(fd, chunk_name);
+		} else if (memcmp(chunk_name, "sRGB", 4) == 0) {
+			read_chunk(fd, chunk_name);
+		} else if (memcmp(chunk_name, "sTER", 4) == 0) {
+			read_chunk(fd, chunk_name);
+		} else if (memcmp(chunk_name, "tEXt", 4) == 0) {
+			read_chunk(fd, chunk_name);
+		} else if (memcmp(chunk_name, "tIME", 4) == 0) {
+			read_chunk(fd, chunk_name);
+		} else if (memcmp(chunk_name, "tRNS", 4) == 0) {
+			read_chunk(fd, chunk_name);
+		} else if (memcmp(chunk_name, "zTXt", 4) == 0) {
+			read_chunk(fd, chunk_name);
+		} else {
+			if (flag_IDAT == 1) {
+				printf("Warning : IDAT was successfull decoded, but IEND not find\n");
+				return 0;
+			}
+			printf("Error : invalid png file.\n");
+			free(png_data->image);
+			return -1;
+		}
+		pos += 12 + (int)htonl(chunk_length);
+	}
+	return 0;
+}
+
+int png_load(char *file_name, struct png *png_data) {
```

Comment:
Please, add check for arguments that they are not NULL

Reply:
done